dotnet/src/SemanticKernel.Abstractions/Contents/ChatMessageContent.cs

```diff
@@ -18,20 +22,94 @@ public class ChatMessageContent : KernelContent
     public AuthorRole Role { get; set; }
 
     /// <summary>
-    /// Content of the message
+    /// A convenience property to get or set the text of the first item in the <see cref="Items" /> collection.
     /// </summary>
-    public string? Content { get; set; }
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public string? Content
+    {
+        get
+        {
+            if (this.Items.Count == 0)
+            {
+                return null;
+            }
+
+            if (this.Items[0] is TextContent textContent)
```

Comment:
I think this should be changed to FirstOrDefault or SingleOrDefault. What if first will be image and second TextContent? Then this property will fail.

Reply:
Thanks @Krzysztof318. I received the same feedback during the design review, and it makes sense. I am more inclined to change the code to use SingleOrDefault, as it will prevent cases in which the getter returns different content from the one initially provided through the setter when there is more than one text content item in the Items collection.

Reply:
If it's going to search through all the items for one that's text, it seems a bit arbitrary to then fail if there are multiple that are text, no?

Reply:
Right, alternatively, we can always set and get the content of the first text item, regardless of its position and the number of text items in the Items collection. This way, our solution is more flexible, as the 'Content' property shortcut continues to work and does not explode if there are more than one item in the collection.